pkg/nsx/services/securitypolicy/builder.go

```diff
@@ -1372,17 +1534,165 @@ func (service *SecurityPolicyService) updatePeerExpressions(obj *v1alpha1.Securi
 			}
 		}
 
-		// Since cluster is set as "Segment" or "SegmentPort" memberType, So the final produced group criteria is always treated as a mixed criteria
-		totalCriteriaCount, totalExprCount, err = service.validateSelectorExpressions(
-			matchLabelsCount,
-			matchExpressionsCount,
-			opInValueCount,
-			true,
-		)
+		if isVpcEnable && !groupShared {
+			// In VPC level, it doesn't support mixed expression criteria
+			totalCriteriaCount, totalExprCount, err = service.validateSelectorExpressions(
+				matchLabelsCount,
+				matchExpressionsCount,
+				opInValueCount,
+				false,
+			)
+		} else {
+			// Since cluster memberType is set as "Segment" or "SegmentPort", So the final produced group criteria is always treated as a mixed criteria
+			totalCriteriaCount, totalExprCount, err = service.validateSelectorExpressions(
+				matchLabelsCount,
+				matchExpressionsCount,
+				opInValueCount,
+				true,
+			)
```

Comment:
So if `isVpcEnable` is `true` and `groupShared` is `true`, the logic also will go into L1546, is it expected?

Reply:
yes, It's expected. In this case, which means is's a shared group.So, it will be created in project level, so we will create group criteria with mixed expression using "Segment" and "SegmentPort".